sdk/core/src/services/registeringHandler.ts

```diff
@@ -0,0 +1,94 @@
+  didMethod: string,
+  password: string,
+  platformCryptographyTools: IPlatformCryptographyTools,
+  keyOptions?: KeyOptions,
+) => {
+  const seedWithMethod = await generateFullSeed(platformCryptographyTools, didMethod, keyOptions)
+  const { keysService, encryptedSeed } = await KeysService.fromSeedAndPassword(seedWithMethod, password)
+
+  const didDocumentService = DidDocumentService.createDidDocumentService(keysService)
+  const didDocument = await didDocumentService.buildDidDocument(didResolver)
+  const did = didDocument.id
+  const didDocumentKeyId = didDocumentService.getKeyId()
+
+  await anchorDid(api, encryptedSeed, password, didDocument, false)
+
+  return { did, didDocumentKeyId, encryptedSeed }
+}
+
+const registerElemAnchored = async (
+  api: RegistryApiService,
+  didResolver: DidResolver,
+  password: string,
+  platformCryptographyTools: IPlatformCryptographyTools,
+  keyOptions?: KeyOptions,
+) => {
+  const seedWithHexMethod = await generateSeedHexWithMethod(ELEM_DID_METHOD)
+  const { encryptedSeed, keysService } = await KeysService.fromSeedAndPassword(seedWithHexMethod, password)
+
+  const didElemDocumentService = DidDocumentService.createDidDocumentService(keysService)
+  const didDocument = await didElemDocumentService.buildDidDocument(didResolver)
+
+  const { did } = await anchorDid(api, encryptedSeed, password, didDocument, true)
+
+  const elemAnchoredSeed = await buildElemAnchoredSeed(did, seedWithHexMethod, platformCryptographyTools, keyOptions)
+
+  const {
+    keysService: elemAnchoredKeysService,
+    encryptedSeed: elemAnchoredEncryptedSeed,
+  } = await KeysService.fromSeedAndPassword(elemAnchoredSeed, password)
+  const didDocumentService = DidDocumentService.createDidDocumentService(elemAnchoredKeysService)
+  const didDocumentKeyId = didDocumentService.getKeyId()
+
+  return { did, encryptedSeed: elemAnchoredEncryptedSeed, didDocumentKeyId }
+}
+
+const buildElemAnchoredSeed = async (
+  did: string,
+  originalSeedWithMethod: string,
+  platformCryptographyTools: IPlatformCryptographyTools,
+  keyOptions?: KeyOptions,
+) => {
+  const base64EncodedAdditionalData = await buildBase64EncodedAdditionalData(platformCryptographyTools, keyOptions, {
+    anchoredDid: did,
+  })
+  const seedWithMethod = originalSeedWithMethod.replace('++elem', '++elem-anchored')
+  return joinSeedWithMethodAndBase64EncodedData(seedWithMethod, base64EncodedAdditionalData)
+}
+
+export const register = (
+  api: RegistryApiService,
+  didResolver: DidResolver,
+  didMethod: DidMethod,
+  platformCryptographyTools: IPlatformCryptographyTools,
+  password: string,
+  keyOptions?: KeyOptions,
+) => {
+  if ([ELEM_DID_METHOD, JOLO_DID_METHOD].includes(didMethod)) {
+    return registerJoloOrElem(api, didResolver, didMethod, password, platformCryptographyTools, keyOptions)
+  }
+
+  if (ELEM_ANCHORED_DID_METHOD === didMethod) {
+    return registerElemAnchored(api, didResolver, password, platformCryptographyTools, keyOptions)
```

Comment:
I don't quite understand your explanation, could you please add a commit for that, if you really think it should be included and this update could stop this PR from merging

Reply:
OK, I'll do that.

Reply:
(done)

Reply:
Including tests errors, I expected that, and again it requires changing other files => make PR bigger => harder to review and finding consensus.

Reply:
You said that you face the same test errors in your integration tests (which are not yet pushed).
This will be solved once we merge #149 which will happen any minute now.

> again it requires changing other files

Unfortunately git didn't detect that, but the main change was moving almost all of `ElemDidDocument` to `ElemDidDocumentBuilder`. Although now that I think of it, I see a way of fixing it.

> make PR bigger => harder to review

It makes `registeringHandler` much easier to review.

Reply:
@anton-iskryzhytskyi , replaced the last commit, now git realizes that `ElemDidDocumentBuilder` is just renamed `ElemDidDocument` and there are less changes.